### PR TITLE
Node: Base watcher support for testnet

### PR DESCRIPTION
**Pull request by bruce-riley**

### CODE REVIEW

File: node/cmd/guardiand/node.go [hunk 1]
```diff
@@ -639,13 +646,25 @@ func runNode(cmd *cobra.Command, args []string) {
 		if *neonContract == "" {
 			logger.Fatal("Please specify --neonContract")
 		}
+		if *baseRPC == "" {
+			logger.Fatal("Please specify --baseRPC")
+		}
```
Comment (panoel) [inline node/cmd/guardiand/node.go:651]: Please add the following to node/cmd/guardiand/adminserver.go::DumpRPCs() (line 779)
```
rpcMap["baseRPC"] = *baseRPC

**Review by panoel — APPROVED** [review]

File: node/pkg/watchers/evm/watcher.go [hunk 2]
```diff
@@ -316,6 +316,26 @@ func (w *Watcher) Run(ctx context.Context) error {
...
+			p2p.DefaultRegistry.AddErrorCount(w.chainID, 1)
+			return fmt.Errorf("dialing eth client failed: %w", err)
+		}
+		pollConnector, err := connectors.NewBlockPollConnector(ctx, baseConnector, finalizers.NewDefaultFinalizer(), 250*time.Millisecond, true, true)
+		if err != nil {
+			ethConnectionErrors.WithLabelValues(w.networkName, "dial_error").Inc()
+			p2p.DefaultRegistry.AddErrorCount(w.chainID, 1)
+			return fmt.Errorf("creating block poll connector failed: %w", err)
+		}
+		// Use the Arbitrum connector to get the TimeOfBlockByHash() implementation.
```
Comment (evan-gray) [inline node/pkg/watchers/evm/watcher.go:332]: Maybe we should consider a better name for this connector 😄

**Review by evan-gray — APPROVED** [review]
Approving, but would happily approve again after @panoel's comment is addressed. 🙂

File: node/cmd/guardiand/node.go (hunk 1 above)
Comment (bruce-riley) [reply node/cmd/guardiand/node.go:651]: Done.

File: node/pkg/watchers/evm/watcher.go (hunk 2 above)
Comment (bruce-riley) [reply node/pkg/watchers/evm/watcher.go:332]: Will open a separate PR to do this when I'm bored.

**Review by panoel — APPROVED** [review]

**Review by evan-gray — APPROVED** [review]
A promise made, a promise kept

**Review by hendrikhofstadt — APPROVED** [review]

File: node/pkg/watchers/evm/watcher.go [hunk 3]
```diff
@@ -316,6 +316,26 @@ func (w *Watcher) Run(ctx context.Context) error {
...
 			return fmt.Errorf("failed to create polygon connector: %w", err)
 		}
+	} else if w.chainID == vaa.ChainIDBase && !w.unsafeDevMode {
+		baseConnector, err := connectors.NewEthereumConnector(timeout, w.networkName, w.url, w.contract, logger)
+		if err != nil {
+			ethConnectionErrors.WithLabelValues(w.networkName, "dial_error").Inc()
+			p2p.DefaultRegistry.AddErrorCount(w.chainID, 1)
+			return fmt.Errorf("dialing eth client failed: %w", err)
+		}
+		pollConnector, err := connectors.NewBlockPollConnector(ctx, baseConnector, finalizers.NewDefaultFinalizer(), 250*time.Millisecond, true, true)
```
Comment (hendrikhofstadt) [inline node/pkg/watchers/evm/watcher.go:326]: Why is the pollconnector config different from arbitrum?

Comment (evan-gray) [reply node/pkg/watchers/evm/watcher.go:326]: Base is OP stack (Optimism) which is the Bedrock release, and so it support `safe` and `finalized` tags. I believe that's why it matches Eth in this way.

Comment (bruce-riley) [reply node/pkg/watchers/evm/watcher.go:326]: That is correct. Discussed this with @hendrikhofstadt offline. I think we are good on this.

**Pull request merged**